geth/log/log.go

```diff
@@ -2,16 +2,28 @@ package log
 
 import (
 	"fmt"
+	"os"
 	"strings"
 
 	"github.com/ethereum/go-ethereum/log"
 )
 
-// logger is package scope instance of log.Logger
-var logger = log.New("geth", "StatusIM")
+// Logger is a wrapper around log.Logger.
+type Logger struct {
+	log.Logger
+	Level   log.Lvl
+	Handler log.Handler
+}
+
+// logger is package scope instance of Logger
+var logger = Logger{
+	Logger:  log.New("geth", "StatusIM"),
+	Level:   log.LvlInfo,
+	Handler: log.StdoutHandler,
+}
 
 func init() {
```

Comment:
Can we reduce usage of `init()` ?
In `main.go` there is a func `app.Before = func(ctx *cli.Context) error` which we can use to init a logging with flags.

Reply:
Yeah, I hate `init()` magic as well, and using both `init()` and var literal for initializing also looks too much. But main idea is that package `geth/log` should be usable without any additional calls.

Just setting fields `Level` and `Handler` to default values is not enough, they used only to store log state between `SetLevel()` and `SetLogFile()` calls. By default, log have `DiscardHandler` handler, which means no logging at all. Somewhere in the package  `setHandler()` call  should be made at the initialization.

Calling it from `app.Before` is not a problem, thanks for suggestion - I like it much more than initializing log from `makeNodeConfig()`. But it still doesn't solve the "make log package ready with sane defaults" question. Any package that imports `geth/log` should expect that log initialized and working with sane defaults. Does it make sense?

Reply:
The package `go-ethereum/log` will take care of default values. Any other logging options (log level, log file) will come from flags that will override it. 
As I remeber, old log package had defaults written to file, which can cause issues on low space devices. The logic of log package is that we use default values (Stdout) and if we need to debug and save output, we will then store it by overriding values.
This can come in handy when we switch to production mode, to disabling storing of the log.